src/Filename.php

```diff
@@ -53,6 +53,27 @@ public function exists() {
         return file_exists($this->name);
     }
 
+    /**
+     * @return bool
+     */
+    public function isWritable() {
+        if (is_writeable($this->asString())) {
+            return true;
+        }
+
+        if (is_link($this->asString())) {
+            // file_exists follows symlinks and returns false if the linked file does not exist
+            // so we need to check first if the file is a link
+            return false;
```

Comment:
I'm not convinced this is technically correct here.

I understand that, if it's a link and points to an existing file that happens to be writable, the first check with catch that. 

But: If the link points to a non existing file, trying to write to it anyhow will simply create the missing end - if the end can be written to.

Reply:
> But: If the link points to a non existing file, trying to write to it anyhow will simply create the missing end - if the end can be written to.
That's not correct, because if it is a link it is considered non writable (the method returns false and the writable check above failed already). So an Exception will be thrown and the file will not be written (at least with Phive - I am not sure if you see this package as a standalone library for other projects).

I am not sure, if this is the perfect behavior, but in any case there will be a reasonable error message and the user is able to check the existing file / link.

By default this shouldn't happen anyway. In my case I copied my local files into a Docker image where the link was invalid afterwards.

Probably we can check the owner of the file with `stat()` and return true if it is owned by the current user (which is save, because the link / file is removed before Phive writes a new link / file). However, on Windows stat() gives us only 0 as user and group, so there needs to be some special handling. And I am not sure if the extra complexity is worth it.